release/github/publish-github-release.ts

```diff
@@ -0,0 +1,57 @@
+import { writeFile } from 'node:fs/promises'
+import { join } from 'node:path'
+import { createReleaseNote } from './create-release-note'
+import { checkIsNewRelease } from '../utils'
+import { getPackageChangelogInfo } from './get-package-changelog-info'
+import { getCredits } from './get-credits'
+
+async function publishGitHubRelease() {
+  const { isDryRun } = checkIsNewRelease()
+  const changelogs = getPackageChangelogInfo()
+  const credits = await getCredits()
+  const releaseNote = createReleaseNote(changelogs, credits)
+  const nextjsVersion = changelogs['next'].version
+  const isCanary = nextjsVersion.includes('canary')
+  try {
+    // https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28#create-a-release
+    const res = await fetch(
```

Comment:
should we use octokit for interacting with GitHub's APIs? I think we already have it in some of the other workflows.